chia/full_node/full_node.py

```diff
@@ -716,21 +719,20 @@ async def new_peak(self, request: full_node_protocol.NewPeak, peer: ws.WSChiaCon
 
         if self.sync_store.get_sync_mode():
             # If peer connects while we are syncing, check if they have the block we are syncing towards
-            peak_sync_hash = self.sync_store.get_sync_target_hash()
-            peak_sync_height = self.sync_store.get_sync_target_height()
-            if peak_sync_hash is not None and request.header_hash != peak_sync_hash and peak_sync_height is not None:
-                peak_peers: Set[bytes32] = self.sync_store.get_peers_that_have_peak([peak_sync_hash])
+            target_peak = self.sync_store.target_peak
+            if target_peak is not None and request.header_hash != target_peak.header_hash:
```

Comment:
It would be good to have a test or invariant assert that ensures that if either `height` or `header_hash` was None in the old code, that `target_peak` is always None is the new code.

Reply:
There were either both `None` or none of them was `None` since they were initially `None` and only set via https://github.com/Chia-Network/chia-blockchain/blob/bf4759be503bf46c997b41e39a3c22a53316e404/chia/full_node/sync_store.py#L41-L43 which sets both. Now they are just packed into `Peak` so the behaviour is the same, isn't it? Either `target_peak` is non, which would be the same as both of them being `None` before or `target_peak` is not non and then both of them are also not `None` because they are supposed to be `bytes32` and `uint32` by their definition.

Reply:
having hash None but hight not None or the other way around doesn't make sense 
if i had to guess the checks in the old code are only for lint purposes.

i think the new code is better because it enforces that fact that if we know the peak than we must know its hash and its hight and the wight.

Reply:
Yes, agreed. I noticed the same thing. This way covers more cases of possible future misuse. :)